formats/cbor/commonTest/src/kotlinx/serialization/cbor/CborReaderTest.kt

```diff
@@ -104,4 +108,481 @@ class CborReaderTest {
+            expect("a")
+            skipElement() // [1, 255, 65536]
+            expect("b")
+            skipElement() // {"x": "kotlinx", "y": "serialization"}
+            expect("c")
+            expect(0)
+        }
+    }
+
+    /**
+     * Tests skipping unknown keys associated with **indefinite length** values of the following CBOR types:
+     * - Major type 2: a byte string
+     * - Major type 3: a text string
+     * - Major type 4: an array of data items
+     * - Major type 5: a map of pairs of data items
+     */
+    @Test
+    fun testSkipIndefiniteLength() {
+        /* A5                                  # map(5)
+         *    61                               # text(1)
+         *       61                            # "a"
+         *    5F                               # bytes(*)
+         *       42                            # bytes(2)
+         *          CAFE                       # "\xCA\xFE"
+         *       43                            # bytes(3)
+         *          010203                     # "\x01\x02\x03"
+         *       FF                            # primitive(*)
+         *    61                               # text(1)
+         *       62                            # "b"
+         *    7F                               # text(*)
+         *       66                            # text(6)
+         *          48656C6C6F20               # "Hello "
+         *       65                            # text(5)
+         *          776F726C64                 # "world"
+         *       FF                            # primitive(*)
+         *    61                               # text(1)
+         *       63                            # "c"
+         *    9F                               # array(*)
+         *       67                            # text(7)
+         *          6B6F746C696E78             # "kotlinx"
+         *       6D                            # text(13)
+         *          73657269616C697A6174696F6E # "serialization"
+         *       FF                            # primitive(*)
+         *    61                               # text(1)
+         *       64                            # "d"
+         *    BF                               # map(*)
+         *       61                            # text(1)
+         *          31                         # "1"
+         *       01                            # unsigned(1)
+         *       61                            # text(1)
+         *          32                         # "2"
+         *       02                            # unsigned(2)
+         *       61                            # text(1)
+         *          33                         # "3"
+         *       03                            # unsigned(3)
+         *       FF                            # primitive(*)
+         *    61                               # text(1)
+         *       65                            # "e"
+         *    00                               # unsigned(0)
```

Comment:
This trailing data is so that we have something to validate that we skipped the last element (also appears in a few other tests).

Would it be better to add an `isEof` function to `CborDecoder`?
I could use the `isEof` in [`skipElement` function](https://github.com/twyatt/kotlinx.serialization/blob/abb0d5aac1d50bd384215b7f72fb1d9db4169823/formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Encoding.kt#L488), but other than that I had trouble justifying adding a function simply for tests?

Reply:
Since decoder does not have any public api, I think it is ok to add `isEof` to it. Moreover, cases when we skip last element with and without trailing data seem different to me, so it is important to test both.